tide/core/data/FFMPEGMovie.cpp

```diff
@@ -177,59 +166,52 @@ double FFMPEGMovie::getFrameDuration() const
     return _videoStream->getFrameDuration();
 }
 
-TextureFormat FFMPEGMovie::getFormat() const
-{
-    return _format;
-}
-
-void FFMPEGMovie::setFormat(const TextureFormat format)
-{
-    _format = format;
-}
-
 PicturePtr FFMPEGMovie::getFrame(double posInSeconds)
 {
     posInSeconds = std::max(0.0, std::min(posInSeconds, getDuration()));
+    const auto frameDuration = _videoStream->getFrameDuration();
+    const auto target = std::max(0.0, posInSeconds - frameDuration);
+    const auto frameIndex = _videoStream->getFrameIndex(target);
+
+    _frameIndex = frameIndex;
+    _streamPosition = posInSeconds;
+    int64_t frameIndexCurr = _frameLastDecode;
+    _frameLastDecode = _frameIndex;
 
     // Seek back for loop or forward if too far away
-    const auto streamDelta = posInSeconds - _streamPosition;
-    if (streamDelta < 0 || std::abs(streamDelta) > MIN_SEEK_DELTA_SEC)
+    const auto streamDelta = frameIndex - frameIndexCurr;
+    if (streamDelta < 0 || std::abs(streamDelta) > MIN_SEEK_DELTA_FRAMES)
     {
-        const auto frameDuration = _videoStream->getFrameDuration();
-        const auto target = std::max(0.0, posInSeconds - frameDuration);
-        const auto frameIndex = _videoStream->getFrameIndex(target);
         if (!_videoStream->seekToNearestFullframe(frameIndex))
-            return PicturePtr();
+            return nullptr;
     }
 
-    const auto targetTimestamp = _videoStream->getTimestamp(posInSeconds);
+    const auto targetTimestamp = _videoStream->getTimestamp(frameIndex);
```

Comment:
still posInSeconds?

Reply:
There are two getTimestamp methods. One that takes a double(seconds) and one that takes an int(frame). I removed the double one.